### PR TITLE
Add dedicated JDBC regression tests for hasOverlappingSiblings overlap detection

### LINKED ISSUE #4600: JDBC: Add dedicated regression coverage for hasOverlappingSiblings behavior
### Summary
`JdbcBasePersistenceImpl.hasOverlappingSiblings()` is JDBC-specific (schema v2+ overlap query + location comparison). NoSQL already has dedicated behavioral coverage in `TestNoSqlMetaStoreManager.overlappingLocations()`, but JDBC did not have equivalent tests on `main`.
The production bug is tracked in #3378 and addressed in #4580 (fix + initial JDBC regression tests).
This issue tracks **remaining** JDBC regression coverage beyond #4580, not the core fix itself.

### Already addressed by #4580
- Use `PolarisEntityUtils.asLocationBasedEntity(PolarisEntity.of(result))` instead of casting query rows to `LocationBasedEntity`
- Initial JDBC tests in `AtomicMetastoreManagerWithJdbcBasePersistenceImplTest` (overlap + non-overlap scenarios)
- Schema-version execution via `AtomicMetastoreManagerWithJdbcBasePersistenceImplSchemaTest`

### Remaining coverage (follow-up)
Port additional scenarios from `TestNoSqlMetaStoreManager.overlappingLocations()` to JDBC/H2, for example:
- duplicate base locations (same path, trailing-slash variants)
- parent/child path overlap in both directions
- scheme normalization where applicable (`s3` vs `s3a`, etc.)
- behavior after dropping one overlapping entity
Note: `QueryGeneratorTest.testGenerateOverlapQuery` covers overlap **SQL generation** only, not full persistence behavior.

### Related
- #3378 — original bug report
- #4580 — fix + initial regression tests

### PR DESCRIPTION
**Pull request by harshitaajoshi**
## What

Adds comprehensive regression test coverage for `hasOverlappingSiblings` in the JDBC persistence layer, as requested in #4600. Also adds a concrete H2-based parameterized test runner so these tests can run locally without Docker.

## Why

The existing JDBC test only covered a basic parent/child overlap and a non-overlapping case. The NoSQL layer had much broader coverage for edge cases (trailing slashes, both overlap directions, duplicates, entity drops) but the JDBC layer did not. These regression tests ensure the JDBC overlap detection behaves correctly across all important scenarios and prevents future regressions.

Fixes #4600

## Changes

- Added 4 new test methods to `AtomicMetastoreManagerWithJdbcBasePersistenceImplTest`:
  - `testHasOverlappingSiblingsTrailingSlashVariants`: verifies that a location without a trailing slash still overlaps with a stored location that has one
  - `testHasOverlappingSiblingsParentChildBothDirections`: verifies overlap is detected when the candidate is a child of an existing entity AND when it is a parent of an existing entity
  - `testHasOverlappingSiblingsDuplicateLocations`: verifies that an exact duplicate base location is reported as overlapping
  - `testHasOverlappingSiblingsAfterEntityDrop`: verifies overlap persists while duplicates remain, and disappears only after all overlapping entities are dropped
- Added a `createOverlapTestCatalog` helper to reduce boilerplate across new tests
- Created `AtomicMetastoreManagerWithJdbcBasePersistenceImplH2SchemaTest` as a concrete H2 parameterized test runner (mirrors the Postgres IT pattern but runs without Docker)

## Checklist
- [x] Don't disclose security issues! (contact security@apache.org)
- [x] Clearly explained why the changes are needed, or linked related issues: Fixes #4600
- [x] Added/updated tests with good coverage (and explained how)
- [x] Added comments for complex logic
- [ ] Updated `CHANGELOG.md` (if needed)
- [ ] Updated documentation in `site/content/in-dev/unreleased` (if needed)

### CODE REVIEW

**Review by ayushtkn — COMMENTED** [review]
Thanx @harshitaajoshi for volunteering!!! 

Few suggestions here:

Scheme normalization not covered
Issue #4600 lists s3 vs s3a normalization as remaining coverage. The NoSQL test explicitly checks:

List.of("s3://bucket/foo/bar", "s3://bucket/foo/bar/", "s3a://bucket/foo/bar/", ...)
This PR does not add an equivalent JDBC test. StorageLocation/S3Location handles scheme equivalence at comparison time, but a JDBC regression test would still be valuable since the SQL pre-filter uses location_without_scheme.

Existing test could use the new helper
`testHasOverlappingSiblingsUsesStoredBaseLocation` still inlines catalog creation. Refactoring it to use `createOverlapTestCatalog` would improve consistency

File: persistence/relational-jdbc/src/test/java/org/apache/polaris/persistence/relational/jdbc/AtomicMetastoreManagerWithJdbcBasePersistenceImplTest.java [hunk 1]
```diff
@@ -158,6 +159,205 @@ void testHasOverlappingSiblingsUsesStoredBaseLocation() {
...
+            catalog,
+            metaStoreManager.generateNewEntityId(callContext).getId(),
+            "drop_ns2",
+            "s3://bucket/drop/path/");
+    metaStoreManager.createEntityIfNotExists(
+        callContext, List.of(PolarisEntity.toCore(catalog)), ns2);
+
+    // Drop the first entity
+    metaStoreManager.dropEntityIfExists(
+        callContext, List.of(PolarisEntity.toCore(catalog)), ns1, null, false);
```
Comment (flyingImer) [inline persistence/relational-jdbc/src/test/java/org/apache/polaris/persistence/relational/jdbc/AtomicMetastoreManagerWithJdbcBasePersistenceImplTest.java:321]: Could we assert that both drops succeed? If a regression caused the second write at this location to replace ns1, the first drop would return ENTITY_NOT_FOUND, but both overlap assertions would still pass with only ns2 present initially. Checking the drop results would make this test catch that regression as well.

**Review by flyingImer — APPROVED** [review]
No blocking concerns from my side. One non-blocking suggestion inline to strengthen the regression test. I also agree with @ayushtkn on adding the scheme coverage before closing #4600.
